lib/LaTeXML/Core/Whatsit.pm

```diff
@@ -218,6 +218,31 @@ sub beAbsorbed {
   LaTeXML::Core::Definition::stopProfiling($profiled, 'absorb') if $profiled;
   return @result; }
 
+# Similar to ->revert, but converts to pure string for use in an attribute value
+sub toAttribute {
+  my ($self) = @_;
+  my $props  = $$self{properties};
+  my $defn   = $self->getDefinition;
+  my $spec   = $$props{toAttribute} || $$defn{toAttribute};
+  if (!defined $spec) {
+    return $self->toString; }    # Default
+  elsif (ref $spec eq 'CODE') {    # If handled by CODE, call it
+    $spec = &$spec($self, $self->getArgs); }
+  # Now, similar to substituteParameters, but creating a string.
+  $spec =~ s/#(#|[1-9]|\w+)/ toAttribute_aux($self,$1)/eg;
```

Comment:
Hm, I think I understand the code here, but I am wondering if we should have a an example DefConstructor that uses this argument-based `toAttribute` string form, and a test that exercises it.